mailbox/plugin/deleted-messages-vault/src/main/java/org/apache/james/vault/blob/BlobStoreDeletedMessageVault.java

```diff
@@ -131,6 +134,16 @@
         return Mono.from(messageMetadataVault.remove(bucketName, user, messageId));
     }
 
+    private Mono<Void> deleteSameBlobInOtherBuckets(DeletedMessageWithStorageInformation deletingInformation) {
+        BucketName deletedBlobBucket = deletingInformation.getStorageInformation().getBucketName();
+        BlobId blobId = deletingInformation.getStorageInformation().getBlobId();
+
+        return Flux.from(messageMetadataVault.listRelatedBuckets())
+            .filter(bucket -> !bucket.equals(deletedBlobBucket))
```

Comment:
why do we filter on that?

Reply:
because the blob in `deletedBlobBucket` is already deleted in the first step. Then I dont need to do it again